### PR TITLE
fix: empty list data may be possible for TopLevel

### PR DESCRIPTION
**Pull request by gdoumenc**
Empty list may be possible in data with no error.

In doc:
```
HTTP/1.1 200 OK
Content-Type: application/vnd.api+json

{
  "links": {
    "self": "http://example.com/articles"
  },
  "data": []
}
```

I  think also the data may be null without error but I didn't changed it as I think you may it intentionnaly :

```
HTTP/1.1 200 OK
Content-Type: application/vnd.api+json

{
  "links": {
    "self": "http://example.com/articles/1/author"
  },
  "data": null
}
```
 8.1.1.2 404 Not Found

A server MUST respond with 404 Not Found when processing a request to fetch a single resource that does not exist, except when the request warrants a 200 OK response with null as the primary data (as described above).

### CODE REVIEW

**Comment by impocode** [comment]
Hi! At this point, I'm trying to make sure that I don't have "data" and "errors" keys in the "data" object at the same time. I think it's more correct to do :

```python
if "data" in data and "errors" in data
```

**Comment by gdoumenc** [comment]
IMO it should be only :

`if "data" in data and data.get("errors"):`

**Comment by impocode** [comment]
> IMO it should be only :
> 
> `if "data" in data and data.get("errors"):`

No) Because in your case, I can write like this:

```python
from jsonapi_pydantic.v1_0 import Error, Resource, TopLevel
data = [
    Resource(
        type="articles",
        id="1",
        attributes={
            "title": "JSON:API paints my bikeshed!",
        }
    )
]
top_level = TopLevel(data=data, errors=None)
print(top_level.model_dump(exclude_unset=True))

Print:
{
    'data': [{'type': 'articles', 'id': '1', 'attributes': {'title': 'JSON:API paints my bikeshed!'}}],
    'errors': None
}
```

This is wrong. Because: `7.1: The members data and errors MUST NOT coexist in the same document.`

File: jsonapi_pydantic/v1_0/resource/resource.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 Id = Optional[str]
 Attributes = Optional[Dict[str, Any]]
-Relationships = Optional[Dict[str, Relationship]]
+Relationships = Optional[Dict[str, Union[Relationship, List[Relationship]]]]
```
Comment (impocode) [inline jsonapi_pydantic/v1_0/resource/resource.py:13]: `7.2.2.2: The value of the relationships key MUST be an object (a “relationships object”).`

Comment (gdoumenc) [reply jsonapi_pydantic/v1_0/resource/resource.py:13]: Oups again :)
Perhaps you should provide an example of an array of [resource identifier objects](https://jsonapi.org/format/#document-resource-identifier-objects) for non-empty to-many relationships.
For dummy guys like me

**Pull request closed without merging**